Prometheus crashes on `panic: runtime error: integer divide by zero`

**What did you do?**
Run Prometheus 

**What did you expect to see?**
Prometheus up and running

**What did you see instead? Under which circumstances?**
Prometheus crashed on 

```
time="2016-12-12T14:46:00Z" level=info msg="Collision detected for fingerprint 1e96a5019c8e7521, metric etcd_rafthttp_message_sent_latency_microseconds{instance=\"10.72.4.110:2379\", job=\"etcd\", msgType=\"MsgHeartbeat\", quantile=\"0.9\", remoteID=\"58d9fb3eff84d3cd\", sendingType=\"pipeline\"}, mapping to new fingerprint 0000000000000016." source="mapper.go:182" 
panic: runtime error: integer divide by zero

goroutine 376 [running]:
panic(0x18939a0, 0xc420018040)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/prometheus/prometheus/storage/local/chunk.doubleDeltaEncodedChunk.Add(0xc43c592c00, 0x287, 0x400, 0x158f37f3864, 0x40569068fcfdb0df, 0x2d0c6e110b2a7f85, 0xc5ef313840, 0x54e34d, 0xc4218ed9c0, 0xc421872d50)
        /go/src/github.com/prometheus/prometheus/storage/local/chunk/doubledelta.go:86 +0xf76
github.com/prometheus/prometheus/storage/local/chunk.(*doubleDeltaEncodedChunk).Add(0xc615c82080, 0x158f37f3864, 0x40569068fcfdb0df, 0xc5ef313940, 0x557434, 0xc4218ed9c0, 0xd7cbbb540abb1586, 0xc5a0bc5650)
        <autogenerated>:36 +0x7e
github.com/prometheus/prometheus/storage/local/chunk.(*Desc).Add(0xc5680d94c0, 0x158f37f3864, 0x40569068fcfdb0df, 0x0, 0x0, 0x2, 0x8020106, 0x2)
        /go/src/github.com/prometheus/prometheus/storage/local/chunk/chunk.go:136 +0x4a
github.com/prometheus/prometheus/storage/local.(*memorySeries).add(0xc5a0bc5650, 0x158f37f3864, 0x40569068fcfdb0df, 0xc5a0bc5650, 0x0, 0x0)
        /go/src/github.com/prometheus/prometheus/storage/local/series.go:245 +0x115
github.com/prometheus/prometheus/storage/local.(*MemorySeriesStorage).Append(0xc42011e000, 0xc643f4dfe0, 0x0, 0x0)
        /go/src/github.com/prometheus/prometheus/storage/local/storage.go:858 +0x398
github.com/prometheus/prometheus/storage.Fanout.Append(0xc4201adba0, 0x2, 0x2, 0xc643f4dfe0, 0xc480817d40, 0x26f5670)
        /go/src/github.com/prometheus/prometheus/storage/storage.go:60 +0x66
github.com/prometheus/prometheus/storage.(*Fanout).Append(0xc42043b020, 0xc643f4dfe0, 0xc5ef313b50, 0xc5ef313b40)
        <autogenerated>:3 +0x6e
github.com/prometheus/prometheus/retrieval.ruleLabelsAppender.Append(0x265af40, 0xc42043b020, 0xc48081c3c0, 0xc643f4dfe0, 0x0, 0x0)
        /go/src/github.com/prometheus/prometheus/retrieval/target.go:241 +0x1b3
github.com/prometheus/prometheus/retrieval.(*ruleLabelsAppender).Append(0xc480813220, 0xc643f4dfe0, 0x0, 0x0)
        <autogenerated>:33 +0x6e
github.com/prometheus/prometheus/retrieval.(*scrapeLoop).append(0xc4807fea00, 0xc66c1aa000, 0x4a7e, 0x4c00)
        /go/src/github.com/prometheus/prometheus/retrieval/scrape.go:456 +0x92
github.com/prometheus/prometheus/retrieval.(*scrapeLoop).run(0xc4807fea00, 0x6fc23ac00, 0x6fc23ac00, 0x0)
        /go/src/github.com/prometheus/prometheus/retrieval/scrape.go:425 +0x602
created by github.com/prometheus/prometheus/retrieval.(*scrapePool).sync
        /go/src/github.com/prometheus/prometheus/retrieval/scrape.go:240 +0x3e5
```

**Environment**

* System information:

```
/prometheus # uname -srm
Linux 4.7.3-coreos-r2 x86_64
```

* Prometheus version: 

```
/prometheus # prometheus -version
prometheus, version 1.4.1 (branch: master, revision: 2a89e8733f240d3cd57a6520b52c36ac4744ce12)
  build user:       root@e685d23d8809
  build date:       20161128-09:59:22
  go version:       go1.7.3
```

* Alertmanager version:

N/A

* Prometheus configuration file:
```
    global:
      scrape_interval: 30s
      scrape_timeout: 30s
    rule_files:
    - /etc/prometheus/recording.rules
    scrape_configs:
    - job_name: etcd
      static_configs:
        - targets:
          - 10.72.0.1:2379
          - 10.72.0.2:2379
          - 10.72.0.3:2379
          - 10.72.0.4:2379
          - 10.72.0.5:2379
    - job_name: 'prometheus'
      static_configs:
        - targets: ['localhost:9090']
    - job_name: 'cloudwatch'
      static_configs:
        - targets: ['prometheus-cloudwatch-exporter:80']
    
    - job_name: 'kubernetes-apiservers'
      kubernetes_sd_configs:
      - role: endpoints
      scheme: https
      tls_config:
        ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
        insecure_skip_verify: true
      bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
      relabel_configs:
      - source_labels: [__meta_kubernetes_namespace, __meta_kubernetes_service_name, __meta_kubernetes_endpoint_port_name]
        action: keep
        regex: default;kubernetes;https
    - job_name: 'kubernetes-nodes'
      scheme: https
      tls_config:
        ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
        insecure_skip_verify: true
      bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
      kubernetes_sd_configs:
      - role: node
      relabel_configs:
      - action: labelmap
        regex: __meta_kubernetes_node_label_(.+)
    - job_name: 'kubernetes-service-endpoints'
      scheme: https
      kubernetes_sd_configs:
      - role: endpoints
      relabel_configs:
      - source_labels: [__meta_kubernetes_service_annotation_prometheus_io_scrape]
        action: keep
        regex: true
      - source_labels: [__meta_kubernetes_service_annotation_prometheus_io_scheme]
        action: replace
        target_label: __scheme__
        regex: (https?)
      - source_labels: [__meta_kubernetes_service_annotation_prometheus_io_path]
        action: replace
        target_label: __metrics_path__
        regex: (.+)
      - source_labels: [__address__, __meta_kubernetes_service_annotation_prometheus_io_port]
        action: replace
        target_label: __address__
        regex: (.+)(?::\d+);(\d+)
        replacement: $1:$2
      - action: labelmap
        regex: __meta_kubernetes_service_label_(.+)
      - source_labels: [__meta_kubernetes_service_namespace]
        action: replace
        target_label: kubernetes_namespace
      - source_labels: [__meta_kubernetes_service_name]
        action: replace
        target_label: kubernetes_name
    - job_name: 'kubernetes-services'
      scheme: https
      metrics_path: /probe
      params:
        module: [http_2xx]
      kubernetes_sd_configs:
      - role: service
      relabel_configs:
      - source_labels: [__meta_kubernetes_service_annotation_prometheus_io_probe]
        action: keep
        regex: true
      - source_labels: [__address__]
        target_label: __param_target
      - target_label: __address__
        replacement: blackbox
      - source_labels: [__param_target]
        target_label: instance
      - action: labelmap
        regex: __meta_kubernetes_service_label_(.+)
      - source_labels: [__meta_kubernetes_service_namespace]
        target_label: kubernetes_namespace
      - source_labels: [__meta_kubernetes_service_name]
        target_label: kubernetes_name
    - job_name: 'kubernetes-pods'
      scheme: https
      kubernetes_sd_configs:
      - role: pod
      relabel_configs:
      - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_scrape]
        action: keep
        regex: true
      - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_path]
        action: replace
        target_label: __metrics_path__
        regex: (.+)
      - source_labels: [__address__, __meta_kubernetes_pod_annotation_prometheus_io_port]
        action: replace
        regex: (.+):(?:\d+);(\d+)
        replacement: ${1}:${2}
        target_label: __address__
      - action: labelmap
        regex: __meta_kubernetes_pod_label_(.+)
      - source_labels: [__meta_kubernetes_pod_namespace]
        action: replace
        target_label: kubernetes_namespace
      - source_labels: [__meta_kubernetes_pod_name]
        action: replace
        target_label: kubernetes_pod_name
```

* Alertmanager configuration file:

N/A

* Logs:
```
time="2016-12-12T14:45:52Z" level=info msg="Collision detected for fingerprint 8b31db99ef569973, metric container_cpu_usage_seconds_total{beta_kubernetes_io_arch=\"amd64\", beta_kubernetes_io_instance_type=\"c3.8
xlarge\", beta_kubernetes_io_os=\"linux\", cpu=\"cpu03\", failure_domain_beta_kubernetes_io_region=\"us-west-2\", failure_domain_beta_kubernetes_io_zone=\"us-west-2b\", id=\"/system.slice/systemd-tmpfiles-setup.s
ervice\", instance=\"ip-10-72-21-104.us-west-2.compute.internal\", job=\"kubernetes-nodes\", kubernetes_io_hostname=\"ip-10-72-21-104.us-west-2.compute.internal\", role=\"minion\"}, mapping to new fingerprint 000
0000000000001." source="mapper.go:182" 
time="2016-12-12T14:45:52Z" level=info msg="Collision detected for fingerprint aeb7e8aa589c92cc, metric container_cpu_usage_seconds_total{beta_kubernetes_io_arch=\"amd64\", beta_kubernetes_io_instance_type=\"c3.8
xlarge\", beta_kubernetes_io_os=\"linux\", cpu=\"cpu05\", failure_domain_beta_kubernetes_io_region=\"us-west-2\", failure_domain_beta_kubernetes_io_zone=\"us-west-2b\", id=\"/system.slice/download-socat-binary.se
rvice\", instance=\"ip-10-72-21-104.us-west-2.compute.internal\", job=\"kubernetes-nodes\", kubernetes_io_hostname=\"ip-10-72-21-104.us-west-2.compute.internal\", role=\"minion\"}, mapping to new fingerprint 0000
000000000002." source="mapper.go:182" 
...
time="2016-12-12T14:46:00Z" level=info msg="Collision detected for fingerprint 1e96a5019c8e7521, metric etcd_rafthttp_message_sent_latency_microseconds{instance=\"10.72.4.110:2379\", job=\"etcd\", msgType=\"MsgHeartbeat\", quantile=\"0.9\", remoteID=\"58d9fb3eff84d3cd\", sendingType=\"pipeline\"}, mapping to new fingerprint 0000000000000016." source="mapper.go:182" 
panic: runtime error: integer divide by zero

goroutine 376 [running]:
panic(0x18939a0, 0xc420018040)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/prometheus/prometheus/storage/local/chunk.doubleDeltaEncodedChunk.Add(0xc43c592c00, 0x287, 0x400, 0x158f37f3864, 0x40569068fcfdb0df, 0x2d0c6e110b2a7f85, 0xc5ef313840, 0x54e34d, 0xc4218ed9c0, 0xc421872d50)
        /go/src/github.com/prometheus/prometheus/storage/local/chunk/doubledelta.go:86 +0xf76
github.com/prometheus/prometheus/storage/local/chunk.(*doubleDeltaEncodedChunk).Add(0xc615c82080, 0x158f37f3864, 0x40569068fcfdb0df, 0xc5ef313940, 0x557434, 0xc4218ed9c0, 0xd7cbbb540abb1586, 0xc5a0bc5650)
        <autogenerated>:36 +0x7e
github.com/prometheus/prometheus/storage/local/chunk.(*Desc).Add(0xc5680d94c0, 0x158f37f3864, 0x40569068fcfdb0df, 0x0, 0x0, 0x2, 0x8020106, 0x2)
        /go/src/github.com/prometheus/prometheus/storage/local/chunk/chunk.go:136 +0x4a
github.com/prometheus/prometheus/storage/local.(*memorySeries).add(0xc5a0bc5650, 0x158f37f3864, 0x40569068fcfdb0df, 0xc5a0bc5650, 0x0, 0x0)
        /go/src/github.com/prometheus/prometheus/storage/local/series.go:245 +0x115
github.com/prometheus/prometheus/storage/local.(*MemorySeriesStorage).Append(0xc42011e000, 0xc643f4dfe0, 0x0, 0x0)
        /go/src/github.com/prometheus/prometheus/storage/local/storage.go:858 +0x398
github.com/prometheus/prometheus/storage.Fanout.Append(0xc4201adba0, 0x2, 0x2, 0xc643f4dfe0, 0xc480817d40, 0x26f5670)
        /go/src/github.com/prometheus/prometheus/storage/storage.go:60 +0x66
github.com/prometheus/prometheus/storage.(*Fanout).Append(0xc42043b020, 0xc643f4dfe0, 0xc5ef313b50, 0xc5ef313b40)
        <autogenerated>:3 +0x6e
github.com/prometheus/prometheus/retrieval.ruleLabelsAppender.Append(0x265af40, 0xc42043b020, 0xc48081c3c0, 0xc643f4dfe0, 0x0, 0x0)
        /go/src/github.com/prometheus/prometheus/retrieval/target.go:241 +0x1b3
github.com/prometheus/prometheus/retrieval.(*ruleLabelsAppender).Append(0xc480813220, 0xc643f4dfe0, 0x0, 0x0)
        <autogenerated>:33 +0x6e
github.com/prometheus/prometheus/retrieval.(*scrapeLoop).append(0xc4807fea00, 0xc66c1aa000, 0x4a7e, 0x4c00)
        /go/src/github.com/prometheus/prometheus/retrieval/scrape.go:456 +0x92
github.com/prometheus/prometheus/retrieval.(*scrapeLoop).run(0xc4807fea00, 0x6fc23ac00, 0x6fc23ac00, 0x0)
        /go/src/github.com/prometheus/prometheus/retrieval/scrape.go:425 +0x602
created by github.com/prometheus/prometheus/retrieval.(*scrapePool).sync
        /go/src/github.com/prometheus/prometheus/retrieval/scrape.go:240 +0x3e5
```

--- a/storage/local/chunk/chunk_test.go
+++ b/storage/local/chunk/chunk_test.go
@@ -39,7 +39,10 @@ func TestLen(t *testing.T) {
 				t.Errorf("chunk type %s should have %d samples, had %d", c.Encoding(), i, c.Len())
 			}
 
-			cs, _ := c.Add(model.SamplePair{model.Time(i), model.SampleValue(i)})
+			cs, _ := c.Add(model.SamplePair{
+				Timestamp: model.Time(i),
+				Value:     model.SampleValue(i),
+			})
 			c = cs[0]
 		}
 	}

--- a/storage/local/chunk/delta.go
+++ b/storage/local/chunk/delta.go
@@ -238,27 +238,36 @@ func (c *deltaEncodedChunk) Unmarshal(r io.Reader) error {
 	if _, err := io.ReadFull(r, *c); err != nil {
 		return err
 	}
-	l := binary.LittleEndian.Uint16((*c)[deltaHeaderBufLenOffset:])
-	if int(l) > cap(*c) {
-		return fmt.Errorf("chunk length exceeded during unmarshaling: %d", l)
-	}
-	if int(l) < deltaHeaderBytes {
-		return fmt.Errorf("chunk length less than header size: %d < %d", l, deltaHeaderBytes)
-	}
-	*c = (*c)[:l]
-	return nil
+	return c.setLen()
 }
 
 // UnmarshalFromBuf implements chunk.
 func (c *deltaEncodedChunk) UnmarshalFromBuf(buf []byte) error {
 	*c = (*c)[:cap(*c)]
 	copy(*c, buf)
+	return c.setLen()
+}
+
+// setLen sets the length of the underlying slice and performs some sanity checks.
+func (c *deltaEncodedChunk) setLen() error {
 	l := binary.LittleEndian.Uint16((*c)[deltaHeaderBufLenOffset:])
 	if int(l) > cap(*c) {
-		return fmt.Errorf("chunk length exceeded during unmarshaling: %d", l)
+		return fmt.Errorf("delta chunk length exceeded during unmarshaling: %d", l)
 	}
 	if int(l) < deltaHeaderBytes {
-		return fmt.Errorf("chunk length less than header size: %d < %d", l, deltaHeaderBytes)
+		return fmt.Errorf("delta chunk length less than header size: %d < %d", l, deltaHeaderBytes)
+	}
+	switch c.timeBytes() {
+	case d1, d2, d4, d8:
+		// Pass.
+	default:
+		return fmt.Errorf("invalid number of time bytes in delta chunk: %d", c.timeBytes())
+	}
+	switch c.valueBytes() {
+	case d0, d1, d2, d4, d8:
+		// Pass.
+	default:
+		return fmt.Errorf("invalid number of value bytes in delta chunk: %d", c.valueBytes())
 	}
 	*c = (*c)[:l]
 	return nil

--- a/storage/local/chunk/delta_test.go
+++ b/storage/local/chunk/delta_test.go
@@ -32,21 +32,20 @@ func TestUnmarshalingCorruptedDeltaReturnsAnError(t *testing.T) {
 		err error,
 		chunkTypeName string,
 		unmarshalMethod string,
-		badLen int) {
+		expectedStr string,
+	) {
 
 		if err == nil {
-			t.Errorf("Failed to obtain an error when unmarshalling %s (from %s) with corrupt length of %d", chunkTypeName, unmarshalMethod, badLen)
+			t.Errorf("Failed to obtain an error when unmarshalling corrupt %s (from %s)", chunkTypeName, unmarshalMethod)
 			return
 		}
 
-		expectedStr := "header size"
 		if !strings.Contains(err.Error(), expectedStr) {
 			t.Errorf(
-				"'%s' not present in error when unmarshalling %s (from %s) with corrupt length %d: '%s'",
+				"'%s' not present in error when unmarshalling corrupt %s (from %s): '%s'",
 				expectedStr,
 				chunkTypeName,
 				unmarshalMethod,
-				badLen,
 				err.Error())
 		}
 	}
@@ -56,6 +55,7 @@ func TestUnmarshalingCorruptedDeltaReturnsAnError(t *testing.T) {
 		chunkConstructor func(deltaBytes, deltaBytes, bool, int) Chunk
 		minHeaderLen     int
 		chunkLenPos      int
+		timeBytesPos     int
 	}{
 		{
 			chunkTypeName: "deltaEncodedChunk",
@@ -64,6 +64,7 @@ func TestUnmarshalingCorruptedDeltaReturnsAnError(t *testing.T) {
 			},
 			minHeaderLen: deltaHeaderBytes,
 			chunkLenPos:  deltaHeaderBufLenOffset,
+			timeBytesPos: deltaHeaderTimeBytesOffset,
 		},
 		{
 			chunkTypeName: "doubleDeltaEncodedChunk",
@@ -72,6 +73,7 @@ func TestUnmarshalingCorruptedDeltaReturnsAnError(t *testing.T) {
 			},
 			minHeaderLen: doubleDeltaHeaderMinBytes,
 			chunkLenPos:  doubleDeltaHeaderBufLenOffset,
+			timeBytesPos: doubleDeltaHeaderTimeBytesOffset,
 		},
 	}
 	for _, c := range cases {
@@ -89,15 +91,26 @@ func TestUnmarshalingCorruptedDeltaReturnsAnError(t *testing.T) {
 
 		cs[0].MarshalToBuf(buf)
 
+		// Corrupt time byte to 0, which is illegal.
+		buf[c.timeBytesPos] = 0
+		err = cs[0].UnmarshalFromBuf(buf)
+		verifyUnmarshallingError(err, c.chunkTypeName, "buf", "invalid number of time bytes")
+
+		err = cs[0].Unmarshal(bytes.NewBuffer(buf))
+		verifyUnmarshallingError(err, c.chunkTypeName, "Reader", "invalid number of time bytes")
+
+		// Fix the corruption to go on.
+		buf[c.timeBytesPos] = byte(d1)
+
 		// Corrupt the length to be every possible too-small value
 		for i := 0; i < c.minHeaderLen; i++ {
 			binary.LittleEndian.PutUint16(buf[c.chunkLenPos:], uint16(i))
 
 			err = cs[0].UnmarshalFromBuf(buf)
-			verifyUnmarshallingError(err, c.chunkTypeName, "buf", i)
+			verifyUnmarshallingError(err, c.chunkTypeName, "buf", "header size")
 
 			err = cs[0].Unmarshal(bytes.NewBuffer(buf))
-			verifyUnmarshallingError(err, c.chunkTypeName, "Reader", i)
+			verifyUnmarshallingError(err, c.chunkTypeName, "Reader", "header size")
 		}
 	}
 }

--- a/storage/local/chunk/doubledelta.go
+++ b/storage/local/chunk/doubledelta.go
@@ -247,28 +247,36 @@ func (c *doubleDeltaEncodedChunk) Unmarshal(r io.Reader) error {
 	if _, err := io.ReadFull(r, *c); err != nil {
 		return err
 	}
-	l := binary.LittleEndian.Uint16((*c)[doubleDeltaHeaderBufLenOffset:])
-	if int(l) > cap(*c) {
-		return fmt.Errorf("chunk length exceeded during unmarshaling: %d", l)
-	}
-	if int(l) < doubleDeltaHeaderMinBytes {
-		return fmt.Errorf("chunk length less than header size: %d < %d", l, doubleDeltaHeaderMinBytes)
-	}
-
-	*c = (*c)[:l]
-	return nil
+	return c.setLen()
 }
 
 // UnmarshalFromBuf implements chunk.
 func (c *doubleDeltaEncodedChunk) UnmarshalFromBuf(buf []byte) error {
 	*c = (*c)[:cap(*c)]
 	copy(*c, buf)
+	return c.setLen()
+}
+
+// setLen sets the length of the underlying slice and performs some sanity checks.
+func (c *doubleDeltaEncodedChunk) setLen() error {
 	l := binary.LittleEndian.Uint16((*c)[doubleDeltaHeaderBufLenOffset:])
 	if int(l) > cap(*c) {
-		return fmt.Errorf("chunk length exceeded during unmarshaling: %d", l)
+		return fmt.Errorf("doubledelta chunk length exceeded during unmarshaling: %d", l)
 	}
 	if int(l) < doubleDeltaHeaderMinBytes {
-		return fmt.Errorf("chunk length less than header size: %d < %d", l, doubleDeltaHeaderMinBytes)
+		return fmt.Errorf("doubledelta chunk length less than header size: %d < %d", l, doubleDeltaHeaderMinBytes)
+	}
+	switch c.timeBytes() {
+	case d1, d2, d4, d8:
+		// Pass.
+	default:
+		return fmt.Errorf("invalid number of time bytes in doubledelta chunk: %d", c.timeBytes())
+	}
+	switch c.valueBytes() {
+	case d0, d1, d2, d4, d8:
+		// Pass.
+	default:
+		return fmt.Errorf("invalid number of value bytes in doubledelta chunk: %d", c.valueBytes())
 	}
 	*c = (*c)[:l]
 	return nil

--- a/storage/local/noop_storage.go
+++ b/storage/local/noop_storage.go
@@ -45,6 +45,9 @@ func (s *NoopStorage) Querier() (Querier, error) {
 	return &NoopQuerier{}, nil
 }
 
+// NoopQuerier is a dummy Querier for use when Prometheus's local storage is
+// disabled. It is returned by the NoopStorage Querier method and always returns
+// empty results.
 type NoopQuerier struct{}
 
 // Close implements Querier.

--- a/storage/local/storage_test.go
+++ b/storage/local/storage_test.go
@@ -319,31 +319,31 @@ func TestFingerprintsForLabels(t *testing.T) {
 		expected model.Fingerprints
 	}{
 		{
-			pairs:    []model.LabelPair{{"label1", "x"}},
+			pairs:    []model.LabelPair{{Name: "label1", Value: "x"}},
 			expected: fingerprints[:0],
 		},
 		{
-			pairs:    []model.LabelPair{{"label1", "test_0"}},
+			pairs:    []model.LabelPair{{Name: "label1", Value: "test_0"}},
 			expected: fingerprints[:10],
 		},
 		{
 			pairs: []model.LabelPair{
-				{"label1", "test_0"},
-				{"label1", "test_1"},
+				{Name: "label1", Value: "test_0"},
+				{Name: "label1", Value: "test_1"},
 			},
 			expected: fingerprints[:0],
 		},
 		{
 			pairs: []model.LabelPair{
-				{"label1", "test_0"},
-				{"label2", "test_1"},
+				{Name: "label1", Value: "test_0"},
+				{Name: "label2", Value: "test_1"},
 			},
 			expected: fingerprints[5:10],
 		},
 		{
 			pairs: []model.LabelPair{
-				{"label1", "test_1"},
-				{"label2", "test_2"},
+				{Name: "label1", Value: "test_1"},
+				{Name: "label2", Value: "test_2"},
 			},
 			expected: fingerprints[15:20],
 		},

